lib/pages/chat/chat_event_list.dart

```diff
@@ -63,88 +64,106 @@ class ChatEventList extends StatelessWidget {
-                    nextEvent: nextEvent,
-                    controller: controller,
+                    );
+                  }
+                  return Container();
+                }
+
+                // The message at this index:
+                final currentEventIndex = i - 1;
+                final event = controller.timeline!.events[currentEventIndex];
+                final previousEvent = currentEventIndex > 0
+                  ? controller.timeline!.events[currentEventIndex - 1]
+                  : null;
+                final nextEvent = i < controller.timeline!.events.length
+                    ? controller.timeline!.events[currentEventIndex + 1]
+                    : null;
+                return AutoScrollTag(
+                  key: ValueKey(event.eventId),
+                  index: currentEventIndex,
+                  controller: controller.scrollController,
+                  child: event.isVisibleInGui
+                      ? Message(
+                          event,
+                          onSwipe: (direction) => controller.replyAction(replyTo: event),
+                          onInfoTab: controller.showEventInfo,
+                          onAvatarTab: (Event event) => showAdaptiveBottomSheet(
+                            context: context,
+                            builder: (c) => UserBottomSheet(
+                              user: event.senderFromMemoryOrFallback,
+                              outerContext: context,
+                              onMention: () => controller.sendController.text +=
+                                  '${event.senderFromMemoryOrFallback.mention} ',
+                            ),
+                          ),
+                          onSelect: controller.onSelectMessage,
+                          scrollToEventId: (String eventId) => controller.scrollToEventId(eventId),
+                          longPressSelect: controller.selectedEvents.isEmpty,
+                          selected: controller.selectedEvents.any((e) => e.eventId == event.eventId),
+                          timeline: controller.timeline!,
+                          previousEvent: previousEvent,
+                          nextEvent: nextEvent,
+                          controller: controller,
+                        )
+                      : Container(),
+                );
+              },
+              childCount: controller.timeline!.events.length + 2,
+              findChildIndexCallback: (key) =>
+                  controller.findChildIndexCallback(key, thisEventsKeyMap),
+            ),
+          ),
+          ValueListenableBuilder<bool>(
+            valueListenable: controller.showTimestampBtn,
+            builder: (context, showTimestampBtn, child) {
+              return showTimestampBtn
+                ? ChatChipButton(
+                  content: controller
+                    .timeline!.events[controller.timeline!.events.length - 1].originServerTs
```

Comment:
```suggestion
                    .timeline!.events.last.originServerTs
```

Reply:
this part is in WIP it should be the first event shown on the screen